packages/core/data-transfer/src/strapi/providers/local-destination/strategies/restore/entities.ts

```diff
@@ -56,12 +60,13 @@ const createEntitiesWriteStream = (options: IEntitiesRestoreStreamOptions) => {
               }
 
               if (attribute.type === 'dynamiczone') {
-                cType = ({ __component }: { __component: string }) => strapi.getModel(__component);
+                cType = ({ __component }: { __component: Common.UID.Component }) =>
+                  strapi.getModel(__component);
               }
             }
           }
 
-          return cType?.uid;
+          return typeof cType === 'object' ? cType?.uid : undefined;
```

Comment:
I feel like this reads strangely wdyt about:

```
if ('uid' in cType) {
  return cType.uid;
}

return undefined;     
```

Reply:
if cType is undefined you cannot use 'uid' in cType this is invalid TS :)

Reply:
so cType can't be undefined so this works :+1